pkg/cloudprovider/aws/pricing.go

```diff
@@ -52,9 +53,13 @@ type PricingProvider struct {
 	onDemandUpdateTime time.Time
 	onDemandPrices     map[string]float64
 	spotUpdateTime     time.Time
-	spotPrices         map[string]float64
+	spotPrices         map[string]map[string]float64
```

Comment:
it might be good to break this out into a type since it's a little weird to read. 
Alternatively, a comment indicating the mapping might be sufficient.